qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
I am not aware of that being the problem. In the original issue (#799)
the following is reported `The qiime mapping file will lower case
everything but the biom will keep the casing.`. And since all the
processing is done using the file produced by
`_get_minimal_mapping_file`, I think we should be good to go, because
that method does not change the letter casing of the samples.

On (Feb-10-15|14:09), Antonio Gonzalez wrote:
> > +from qiita_db.util import get_mountpoint
> > +from qiita_db.sql_connection import SQLConnectionHandler
> > +from qiita_db.metadata_template import PrepTemplate
> > +
> > +conn_handler = SQLConnectionHandler()
> > +
> > +_id, fp_base = get_mountpoint('templates')[0]
> > +
> > +for prep_template_id in conn_handler.execute_fetchall(
> > +        "SELECT prep_template_id FROM qiita.prep_template"):
> > +    prep_template_id = prep_template_id[0]
> > +    pt = PrepTemplate(prep_template_id)
> > +    study_id = pt.study_id
> > +
> > +    for _, fpt in pt.get_filepaths():
> > +        pt.create_qiime_mapping_file(fpt)
> 
> I think you are right about the mapping file, the issue is that the generated biom files will still not match the sample names, right? Which is also a problem. 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/829/files#r24455785